nema-server/src/prompts/summarize/createSummary.ts

```diff
@@ -4,70 +4,120 @@ import { Configuration, OpenAIApi } from 'openai';
 
 dotenv.config();
 
-// let pinecone: PineconeClient | null = null;
-
-// const initPineconeClient = async () => {
-//   pinecone = new PineconeClient();
-//   await pinecone.init({
-//     environment: process.env.PINECONE_ENVIRONMENT!,
-//     apiKey: process.env.PINECONE_API_KEY!,
-//   });
-// };
-
 const configuration = new Configuration({
   apiKey: process.env.OPENAI_API_KEY,
 });
 
 const openai = new OpenAIApi(configuration);
 
-async function generateSummary(chunk: string | undefined) {
+async function generateSummary(chunk: string | undefined): Promise<string | undefined> {
   const prompt = `${chunk}\n\nTl;dr`;
-  const response = await openai.createCompletion({
-    model: 'text-davinci-003',
-    prompt: prompt,
-    temperature: 0,
-    max_tokens: 130,
-    top_p: 1.0,
-    frequency_penalty: 0.2,
-    presence_penalty: 0.2,
-  });
-  const summary: string | undefined = response.data.choices[0].text?.trim();
-  return summary || '';
+  const maxCharacters = 16000;
+
+  if (!chunk) {
+    return '';
+  }
+  if (chunk.length <= maxCharacters) {
+    const response = await openai.createCompletion({
+      model: 'text-davinci-003',
+      prompt: prompt,
+      temperature: 0,
+      max_tokens: 600,
+      top_p: 1.0,
+      frequency_penalty: 0.2,
+      presence_penalty: 0.2,
+    });
+    const summary: string | undefined = response.data.choices[0].text?.trim();
+    return summary || '';
+  } else {
+    const splitIndex = chunk.lastIndexOf(' ', maxCharacters);
```

Comment:
This should not be hardcoded. Like what happens if the chunk is 
* '43000' characters long? 
* '25000' characters long?
* '59000' characters long?

Reply:
yeah I haven't used that value I should've removed that.

Reply:
I have set the max character limit as mentioned. This function can handle that because the chunks are sliced recursively and converted to summaries if the chunk size is more than 16000 characters. I have tested this with a chunk of size '72000' characters long.